lib/sidekiq/cli.rb

```diff
@@ -235,6 +235,10 @@ def boot_system
           end
```

Comment:
I think this change obviates the above forced eager_load, along with the contortions it requires (for Rails 5.0+, of course).

(Or at least it would, if you still wrapped with the `@app.executor` when `cache_classes` is true.)